### PR TITLE
Set storage of root EBS volume

### PR DESCRIPTION
**Pull request by usamasaqib**
What does this PR do?
---------------------
This PR allows the user to specify the storage size of an ec2 instance.

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

**Review by pducolin — APPROVED** [review]
Looks good ! To be double checked if `SearchAmi` change impacts other PRs currently open

File: aws/ec2/ec2/ebs.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ec2"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type BlockDeviceArgs struct {
```
Comment (vboulineau) [inline aws/ec2/ec2/ebs.go:13]: Why going through all that for a dozen of fields?
Why not just having a simple function like:
```
func convertEbsToBlockDeviceArgs(ebs map[string]string) (*ec2.InstanceEbsBlockDeviceArgs, error) {
  args.Foo = ebs["bar"]
}
```

File: aws/ec2/ec2/vm.go [hunk 2]
```diff
@@ -89,6 +83,25 @@ func NewEC2Instance(e aws.Environment, name, ami, arch, instanceType, keyPair, u
...
+	if ami == "" {
+		ami, err = LatestUbuntuAMI(e, arch)
+		if err != nil {
+			return nil, err
+		}
+		device, err := LatestUbuntuAMIRootDevice(e, arch)
+		if err != nil {
+			return nil, err
+		}
+		rootBlockDevice, err := createNewRootDeviceMapping(device, e.DefaultInstanceStorageSize())
```
Comment (vboulineau) [inline aws/ec2/ec2/vm.go:97]: What's the goal of setting this while the AMI is already `Ubuntu`?
Are you trying to simply change the root disk size? In which case you can simply do:
```
        RootBlockDevice: ec2.InstanceRootBlockDeviceArgs{
            VolumeSize: pulumi.IntPtr(200),
        },
```

**Review by vboulineau — CHANGES_REQUESTED** [review]

File: aws/ec2/ec2/vm.go (hunk 2 above)
Comment (usamasaqib) [reply aws/ec2/ec2/vm.go:97]: That is exactly what I needed! Thanks

**Pull request closed without merging**